com.basistech.m2e.code.quality.checkstyle/src/main/java/com/basistech/m2e/code/quality/checkstyle/MavenPluginConfigurationTranslator.java

```diff
@@ -87,14 +104,41 @@ public boolean isSkip() throws CoreException {
 	}
 
 	public URL getRuleset() throws CheckstylePluginException, CoreException {
-		final URL ruleset = resolveLocation(this.getConfigLocation());
+		String configLocation = this.getConfigLocation();
+		URL ruleset = null;
+		if (CHECKSTYLE_DEFAULT_CONFIG_FILE_NAME.equals(configLocation)) {
```

Comment:
@Serranya what is this `CHECKSTYLE_DEFAULT_CONFIG_FILE_NAME` exactly? Is it a fake name used to detect there is no configuration file specified? I find it strange… what happens if the user actually wants to use this file `config/sun_checks.xml`? There will be some problem then, no?

Reply:
The maven-checkstyle-plugin has the following logic:

1. Check if there are inline rules.
2. If there are inline rules check if the config is the `sun_checks.xml` If it is. Use the inline rules. if it isn't throw an error instructing the user to use either the inline rules OR the configFile

Reply:
hm, well, if it matches the behaviour of the maven-checkstyle-plugin it should be alright I guess…

Reply:
Maybe. I am not sure.

For reference here is the code form the maven plugin:
https://github.com/apache/maven-checkstyle-plugin/blob/c4505ee038390f7f2ae90b441a0e0bbe963340d9/src/main/java/org/apache/maven/plugins/checkstyle/CheckstyleViolationCheckMojo.java#L482-L492

If you think throwing an exception is the right thing to do in that case I will change the code accordingly. Im 50/50 on this.

Reply:
I implemented the handling from the maven plugin.

Reply:
I think it's important that m2e-code-quality doesn't behave in a way different from running maven in the command line. So I suppose it is ok to throw this exception but I'm not sure it is the proper way to report errors within an eclipse plugin… @vorburger @erwint?

Reply:
IIRC exceptions during project configuration result in an error dialog and an error marker with the Exception on the POM. So that would be the equivalent to Maven exiting the build with an error.